### PR TITLE
Change diff method from word-based to character-based

### PR DESCRIPTION
**Pull request by wielandbrendel**
Improves #15

### CODE REVIEW

**Comment by dsp05** [comment]
Thanks for making this change. I was considering it as well since sometimes char diff can be significantly larger than word diff, which could lead to UI glitches when applying changes. Maybe we could add an option that lets users choose the level (char or word) or automatically select it based on the number of diffs. For example, use char diff if it is fewer than 100; otherwise, use word diff.

**Comment by wielandbrendel** [comment]
Ah, interesting - did you experience a glitch? I just made a larger change and didn't see anything. But yeah, if it's an issue, letting users select which method to use is a good idea.

**Comment by dsp05** [comment]
For this PR, if it works for you, let's do the auto selection based on the size of the changes. We can check the char diff and apply it if it's fewer than 500. If not, we check the word diff, apply it if that's under 500. Otherwise, we'll replace the entire text as a single change.

**Comment by wielandbrendel** [comment]
Sure, sounds like a great and simple solution! Just implemented the change. I couldn't really test it - somehow, if the chunk of text I select is too large, I don't see the editing tooltip anymore to perform edits.

File: src/utils/actions.ts [hunk 1]
```diff
@@ -16,8 +16,13 @@ export function onAcceptImprovement(
       state.selection.main.from,
       state.selection.main.to
     )
-    const changes = [];
-    const diffs = Diff.diffWordsWithSpace(originalContent, e.detail.improvement);
+    let changes = [];
+    let diffs = Diff.diffChars(originalContent, e.detail.improvement);
+
+    if (diffs.length >= 500) {
+      diffs = Diff.diffWords(originalContent, e.detail.improvement);
```
Comment (dsp05) [inline src/utils/actions.ts:23]: It has to be `diffWordsWithSpace`. `diffWords` ignores some whitespaces and will cause issues when replacing.

**Comment by dsp05** [comment]
> Sure, sounds like a great and simple solution! Just implemented the change. I couldn't really test it - somehow, if the chunk of text I select is too large, I don't see the editing tooltip anymore to perform edits.

Thanks! Let me test it later and see if there are other issues.

File: src/utils/actions.ts (hunk 1 above)
Comment (wielandbrendel) [reply src/utils/actions.ts]: Ah, thanks for catching this! Should be fixed.

**Comment by dsp05** [comment]
Thanks for making the change!

**Pull request merged**